src/test/test-notify-recv.c

```diff
@@ -88,4 +90,44 @@ TEST(notify_socket_prepare) {
         ASSERT_OK(sd_event_loop(e));
 }
 
+TEST(notify_recv) {
+        struct ucred ucred = UCRED_INVALID;
+        _cleanup_close_pair_ int fd[2] = EBADF_PAIR;
+        _cleanup_(pidref_done) PidRef sender = PIDREF_NULL;
+        _cleanup_free_ char *msg = NULL;
+        _cleanup_free_ char *text = NULL;
+
+        ASSERT_OK(socketpair(AF_UNIX, SOCK_DGRAM, 0, fd));
+
+        int optval = 1;
+        ASSERT_OK(setsockopt(fd[1], SOL_SOCKET, SO_PASSCRED, &optval, sizeof(optval)));
+
+        msg = strdup("TEST_MESSAGE=hello");
+        ASSERT_NOT_NULL(msg);
+
+        struct iovec iovec = {
+                .iov_base = msg,
+                .iov_len = strlen(msg) + 1,
+        };
+
+        struct msghdr msghdr = {
+                .msg_iov = &iovec,
+                .msg_iovlen = 1,
+                .msg_control = NULL,
+                .msg_controllen = 0,
+        };
+
+        ssize_t r = sendmsg(fd[0], &msghdr, MSG_NOSIGNAL);
+        ASSERT_EQ((ssize_t)(strlen(msg) + 1), r);
+
+        ASSERT_OK(notify_recv(fd[1], &text, &ucred, &sender));
```

Comment:
notify_recv_strv() used in `on_recv()` in the above is a tiny wrapper of notify_recv().
Hence, unfortunately this does not increase any test coverage.